orchagent/bfdorch.cpp

```diff
@@ -716,6 +722,51 @@ bool BgpGlobalStateOrch::getTsaState()
     SWSS_LOG_ENTER();
     return tsa_enabled;
 }
+
+bool BgpGlobalStateOrch::getSoftwareBfd()
+{
+    SWSS_LOG_ENTER();
+    return !bfd_offload;
+}
+
+bool BgpGlobalStateOrch::offload_supported(bool get_ipv6)
+{
+    sai_attribute_t attr;
+    sai_status_t status;
+    sai_attr_capability_t capability;
+
+    attr.id = SAI_SWITCH_ATTR_SUPPORTED_IPV4_BFD_SESSION_OFFLOAD_TYPE;
+    if(get_ipv6)
+    {
+        attr.id = SAI_SWITCH_ATTR_SUPPORTED_IPV6_BFD_SESSION_OFFLOAD_TYPE;
+    }
+
+    status = sai_query_attribute_capability(gSwitchId, SAI_OBJECT_TYPE_SWITCH,
+                                            attr.id, &capability);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Unable to query BFD offload capability");
+        return false;
+    }
+    if (!capability.set_implemented)
```

Comment:
Should this have been get_implemented? These are read only attributes according to SAI headers

    /**
     * @brief List of BFD session offloads that are supported for IPv4
     *
     * @type sai_s32_list_t sai_bfd_session_offload_type_t
     * @flags READ_ONLY
     */
    SAI_SWITCH_ATTR_SUPPORTED_IPV4_BFD_SESSION_OFFLOAD_TYPE,

    /**
     * @brief List of BFD session offloads that are supported for IPv6
     *
     * @type sai_s32_list_t sai_bfd_session_offload_type_t
     * @flags READ_ONLY
     */
    SAI_SWITCH_ATTR_SUPPORTED_IPV6_BFD_SESSION_OFFLOAD_TYPE,